mqt/syrec/bindings.cpp

```diff
@@ -1,52 +1,259 @@
+            break;
+        case gate_types::v:
+            g.set_type(v_tag());
+            break;
+        case gate_types::vplus:
+            g.set_type(vplus_tag());
+            break;
+        case gate_types::module:
+            g.set_type(module_tag());
+            break;*/
+        default:
+            assert(false);
+            break;
+    }
+}
+
+unsigned gate_get_type(const gate& g) {
+    if (is_toffoli(g)) {
+        return gate_types::toffoli;
+    } else if (is_fredkin(g)) {
+        return gate_types::fredkin;
+    } /*else if (is_peres(g)) {
+        return gate_types::peres;
+    }else if (is_v(g)) {
+        return gate_types::v;
+    } else if (is_vplus(g)) {
+        return gate_types::vplus;
+    } else if (is_module(g)) {
+        return gate_types::module;
+    }*/
+
+    assert(false);
+    return 0u;
+}
+
+/// control and target lines
+
+py::list control_lines1(const gate& g) {
+    gate::line_container c;
+    py::list             l;
+    control_lines(g, std::insert_iterator<gate::line_container>(c, c.begin()));
+    for (const auto& control: c) {
+        l.append(control);
+    }
+    return l;
+}
+
+py::list target_lines1(const gate& g) {
+    gate::line_container c;
+    py::list             l;
+    target_lines(g, std::insert_iterator<gate::line_container>(c, c.begin()));
+    for (const auto& target: c) {
+        l.append(target);
+    }
+    return l;
+}
+
+PYBIND11_MODULE(pysyrec, m) {
+    m.def("py_syrec_synthesis", &syrec_synthesis);
+    m.def("py_simple_simulation", static_cast<bool (*)(boost::dynamic_bitset<>&, const circuit&, const boost::dynamic_bitset<>&, const properties::ptr&, const properties::ptr&)>(simple_simulation));
+
+    py::class_<circuit, std::shared_ptr<circuit>>(m, "circuit")
+            .def(py::init<>())
+            //.def(py::init<unsigned>())
+            .def_property("lines", &circuit::lines, &circuit::set_lines)
+            .def_property_readonly("num_gates", &circuit::num_gates)
+            .def("gates", new_gates)
+            .def_property_readonly("inputs", inputs_getter)
+            .def_property_readonly("outputs", outputs_getter)
+            .def_property_readonly("constants", constants_getter)
+            .def_property_readonly("garbage", garbage_getter)
+            .def("annotations", circuit_annotations);
+
+    m.def("add_new", &add_new, "A function that adds two numbers");
+
+    py::class_<properties, std::shared_ptr<properties>>(m, "properties")
+            .def(py::init<>())
+            .def("set_string", &properties_set<std::string>)
+            .def("set_bool", &properties_set<bool>)
+            .def("set_int", &properties_set<int>)
+            .def("set_unsigned", &properties_set<unsigned>)
+            .def("set_double", &properties_set<double>)
+            .def("set_char", &properties_set<char>)
+            .def("get_string", &properties_get1<std::string>)
+            .def("get_double", &properties_get1<double>);
```

Comment:
Yes, I could do that. I would prefer using different name since it is also suggest to use it that way (https://pybind11.readthedocs.io/en/stable/advanced/functions.html)